docs/sphinx_docs/Input_file_formats/Native/02_types/02_01_builtin.md

```diff
@@ -129,7 +127,73 @@ goal g:
     x + t = 0
 ```
 
-### Bitvectors
+## Additional numeric primitives
+
+Alt-Ergo provides built-in primitives for mixed integers and real problems,
+along with some limited reasoning support, typically restricted to constants.
+These primitives are only available in the native input format.
+
+### Conversion between integers and reals
+
+```alt-ergo
+logic real_of_int : int -> real
+```
+
+`real_of_int` converts an integer into its representation as a real number.
+
+*Note*: When using the `dolmen` frontend, `real_of_int` is also available in
+the smtlib2 format as the `to_real` function from the `Reals_Ints` theory.
+
+```alt-ergo
+logic int_floor : real -> int
+logic int_ceil : real -> int
+logic real_is_int : real -> bool
+```
+
+`int_floor` and `int_ceil` implement the standard `floor` and `ceil` functions.
+They compute the greatest integer less than a real and the least integer
+greater than a real, respectively.
+
+`real_is_int` is true for reals that are exact integers, and false otherwise.
+
+*Note*: When using the Dolmen frontend, `int_floor` and `real_is_int` are
+also available in the smtlib2 format as the `to_int` and `is_int` functions
+from the `Reals_Ints` theory respectively.
+
+### Square root
+
+```alt-ergo
+logic sqrt_real : real -> real
+```
+
+The `sqrt_real` function denotes the square root of a real number.  Alt-Ergo
+only computes rational under- and over-approximations.
```

Comment:
Does it mean `sqrt_real` computes the under-approximation or the over-approximation?

Reply:
`sqrt_real` computes the square root. Internally, Alt-Ergo tracks both an under- and an over- approximation of the square root as rationals. I agree that this sentence is confusing, I will remove it.

Reply:
But I agree we shouldn't write `the square root` without mentioning it's an approximation.

Reply:
It is not an approximation.